include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -380,20 +378,18 @@ class two_dimensional_matrix<value_t, allocator_t, order>::basic_iterator :
     * \param[in] matrix The underlying matrix to access the corresponding dimensions.
     * \param[in] iter   The underlying iterator over the actual storage; must model std::random_access_iterator.
     */
-    constexpr basic_iterator(matrix_t & matrix, storage_iterator iter) :
+    constexpr basic_iterator(parent_t & matrix, storage_iterator iter) :
         matrix_ptr{&matrix},
         host_iter{iter}
     {}
 
     //!\brief Construction of cons-iterator from non-const-iterator.
-    template <typename other_matrix_t>
+    constexpr basic_iterator(basic_iterator<!const_range> const & other) noexcept
     //!\cond
-        requires std::same_as<other_matrix_t, std::remove_const_t<matrix_t>> && std::is_const_v<matrix_t>
+        requires const_range
     //!\endcond
-    constexpr basic_iterator(
-        basic_iterator<other_matrix_t> other) noexcept :
-        matrix_ptr{other.matrix_ptr},
-        host_iter{other.host_iter}
+        : matrix_ptr{other.matrix_ptr},
```

Comment:
Sorry for the :nail_care: but I'm pretty sure we put the `:` after the ctor name?

Reply:
Not sure if you have read the diff completely correct, because it is pretty difficult to read, but the new code looks like this:

```c++
    constexpr basic_iterator(basic_iterator<!const_range> const & other) noexcept
    //!\cond
        requires const_range
    //!\endcond
        : matrix_ptr{other.matrix_ptr},
          host_iter{other.host_iter}
    {}
```

I can't put the `:` at the end of ctor name.

Reply:
ah ok sorry